src/main/java/duke/task/TaskConverter.java

```diff
@@ -0,0 +1,92 @@
+        List<String> stringList = new ArrayList<>();
+        for (Task task : taskList) {
+            String saveLine = task.getType() + "%" + task.isDone() + "%" + task.getName() + "%" + task.getRawTime();
+            stringList.add(saveLine);
+        }
+        return stringList;
+    }
+
+    /**
+     * Converts a list of strings into an ArrayList of Task objects.
+     * Each string is expected to follow the format produced by saveConvert, with task properties separated by "%".
+     *
+     * @param strings A List of strings, each representing a task's persisted data.
+     * @return An ArrayList of Task objects reconstructed from the input strings.
+     */
+    public static ArrayList<Task> loadConvert(List<String> strings) {
+        ArrayList<Task> tasks = new ArrayList<>();
+        try {
+            for (String str : strings) {
+                String[] split = str.split("%");
+                if (split[0].equals("[T]")) {
+                    Task currTask = new Todo(split[2]);
+                    currTask.setDone(Boolean.parseBoolean(split[1]));
+                    tasks.add(currTask);
+                } else if (split[0].equals("[D]")) {
+                    Task currTask = new Deadline(split[2], split[3]);
+                    currTask.setDone(Boolean.parseBoolean(split[1]));
+                    tasks.add(currTask);
+                } else if (split[0].equals("[E]")) {
+                    Task currTask = makeEvent(split[2], split[3]);
+                    currTask.setDone(Boolean.parseBoolean(split[1]));
+                    tasks.add(currTask);
+                }
+            }
+        } catch (Exception e) {
+            System.out.println("Save File Corrupted :(");
+        }
+        return tasks;
+    }
+
+    /**
+     * Helper method to create an Event object from a string containing its name and timeline.
+     *
+     * @param name The name of the event.
+     * @param timeline The timeline of the event, expected to contain start and end times.
+     * @return An Event object constructed from the provided name and timeline.
+     */
+    public static Event makeEvent(String name, String timeline){
+        String[] timeSplit = timeline.split(" ");
+        int toIndex = finder("to:", timeSplit);
+        String startTime = String.join(" ", Arrays.copyOfRange(timeSplit, 1, toIndex));
+        String endTime = String.join(" ", Arrays.copyOfRange(timeSplit, toIndex + 1, timeSplit.length));
+        Event current = new Event(name, startTime, endTime);
+        return current;
+    }
+
+    /**
+     * Searches for a specific string within an array of strings and returns its index.
+     *
+     * @param checker The string to find within the array.
+     * @param list The array of strings to search through.
+     * @return The index of the string if found, or -1 if not found.
+     */
+    public static int finder(String checker, String[] list) {

```

Comment:
might think of renaming checker to something more intuitive such as stringToFind or something similar!